tests/unittests/sources/test_oracle.py

```diff
@@ -636,7 +736,29 @@ def test_retries(
         with expectation:
             assert expected_body == oracle.read_opc_metadata().instance_data
 
+    # No need to actually wait between retries in the tests
+    @mock.patch("cloudinit.url_helper.time.sleep", lambda _: None)
+    def test_fetch_vnics_error(self, caplog):
+        def mocked_fetch(*args, path="instance", **kwargs):
+            if path == "vnics":
+                raise UrlError("cause")
+
+        with mock.patch(DS_PATH + "._fetch", side_effect=mocked_fetch):
+            opc_metadata = oracle.read_opc_metadata(fetch_vnics_data=True)
+            assert None is opc_metadata.vnics_data
+        assert (
+            logging.WARNING,
+            "Failed to fetch IMDS network configuration!",
+        ) == caplog.record_tuples[-2][1:]
 
+
+@pytest.mark.parametrize(
```

Comment:
Nice improvement, but can you ensure this will work on Bionic?

Reply:
The `tox` target `lower-supported-dev` mimics the bionic scenario. After running ```tox -e lowest-supported-dev -- -vv tests/unittests/sources/test_oracle.py```, the same number of tests are executed. I assume there is no problem in Bionic. Are you worried about something specific?